### PR TITLE
Replace parity-hash with fixed-hash

### PR DESCRIPTION
**Pull request by Robbepop**
tl;dr: This PR is the first in a series to clean up the dependency tree of `pwasm-*`.

After a long time finding out which dependencies are deprecated, old, new, replaced etc. I think I am now confident to say that `parity-hash` should be replaced by its successor `fixed-hash`. Since the interface of `fixed-hash` is reduced to be minimal and be comparable to the one provided by `uint` `pwasm-std` has to provide its own hash types now. However, fortunately this isn't a big deal.

I had to rewrite a doc comment test case but now everything works again.
I would recommend releasing a new major version update since the changes from `parity-hash` to `fixed-hash` might break things.

### CODE REVIEW

File: src/types.rs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+construct_hash!(H32, 4);
+construct_hash!(H64, 8);
+construct_hash!(H128, 16);
+construct_hash!(H160, 20);
+construct_hash!(H256, 32);
+construct_hash!(H264, 33);
+construct_hash!(H512, 64);
+construct_hash!(H1024, 128);
+construct_hash!(H2048, 256);
```
Comment (NikVolf) [inline src/types.rs]: probably only H160 and H256 needed?

Comment (Robbepop) [reply src/types.rs]: You are probably right, I just cloned the definitions from `parity-hash`.

Comment (Robbepop) [reply src/types.rs]: I am especially suspicious about the `H264` one. :D

Comment (NikVolf) [reply src/types.rs]: oh, it is used for some signatures - it is ok, just not needed here

**Comment by Robbepop** [comment]
I am currently cleaning up some things in the `fixed-hash` crate and will come back to this PR once that is done.
